common/unsafe/src/main/java/org/apache/spark/unsafe/types/CalendarInterval.java

```diff
@@ -59,6 +59,27 @@ public CalendarInterval subtract(CalendarInterval that) {
     return new CalendarInterval(months, days, microseconds);
   }
 
+  private static CalendarInterval fromDoubles(
+      double monthsWithFraction,
+      double daysWithFraction,
+      double microsWithFraction) {
+    int truncatedMonths = Math.toIntExact((long)(monthsWithFraction));
+    // Using 30 days per month as PostgreSQL does.
+    double days = daysWithFraction + 30 * (monthsWithFraction - truncatedMonths);
```

Comment:
is this consistent with pgsql? i.e. we convert the truncated months to days.

Reply:
yes

Reply:
```sql
maxim=# select interval '1 month'  * 1.1;
   ?column?
--------------
 1 mon 3 days
(1 row)
```